pkgs/os-specific/linux/odp-dpdk/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchgit, autoreconfHook, autoconf, automake, libtool
+, openssl, libpcap, dpdk
+}:
+
+stdenv.mkDerivation rec {
+  name = "odp-dpdk-${version}";
+  version = "1.8.0.0";
+
+  src = fetchgit {
+    url = "https://git.linaro.org/lng/odp-dpdk.git";
+    rev = "438a207a39bad213cdc03929452a8199caef5d8c";
+    sha256 = "0k4g5zbirbfdcgqz0nbn9san66y178qnigyvrr2apj3apzjjy7zv";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+  buildInputs = [ stdenv automake libtool openssl dpdk libpcap ];
+
+  RTE_SDK = "${dpdk}";
+  RTE_TARGET = "x86_64-native-linuxapp-gcc";
+
+  NIX_CFLAGS_COMPILE = [ "-march=core2" ];
+
+  patchPhase = ''
+    substituteInPlace scripts/git_hash.sh --replace /bin/bash /bin/sh
+    echo -n ${version} > .scmversion
+  '';
+
+  configureFlags = [
+    "--with-platform=linux-dpdk"
+    "--enable-static"
+    "--disable-shared"
+    "--disable-shared-dpdk"
```

Comment:
Thanks, I wasn't aware of dontDisableStatic. The reason for static, is some relocation errors when trying to link against statically linked DPDK.

/nix/store/l7bxj0v0wb4mg8fkjb00mdh7ggyw5wfw-binutils-2.26/bin/ld: /nix/store/sdxnfz53vf19ignvbkdhw5nis8y9jr71-dpdk-16.04-4.4.12/lib/librte_pmd_i40e.a(i40e_ethdev.o): relocation R_
X86_64_32 against `.data' can not be used when making a shared object; recompile with -fPIC
/nix/store/sdxnfz53vf19ignvbkdhw5nis8y9jr71-dpdk-16.04-4.4.12/lib/librte_pmd_i40e.a: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
Makefile:797: recipe for target '../../lib/libodp-dpdk.la' failed